src/Authentication.ts

```diff
@@ -7,105 +7,114 @@ import TwoFactorProviderInterface from "./providers/two-factor/TwoFactorProvider
 import RegistrationData from "./types/RegistrationData";
 import LoginData from "./types/LoginData";
 import Session from "./types/Session";
-import TwoFactorAuthenticationData from "./types/TwoFactorAuthenticationData";
 import AuthenticableUser from "./types/AuthenticableUser";
-
-// const postgresConfig: PostgresConnectionData = {
-// 	database: "felony",
-// 	username: "postgres",
-// 	password: "postgrespw",
-// 	host: "localhost",
-// 	port: 55000
-// }
-
-// const postgresConnectionUri = "postgres://postgres:postgrespw@localhost:55000/felony";
+import AuthenticableTwoFactorUser from "./types/AuthenticableTwoFactorUser";
+import ErrorData from "./types/ErrorData";
+import AuthenticationError from "./error/AuthenticationError";
+import { ValidationErrors } from "./error/ValidationError";
 
 export default class Authentication {
+	private errorAdapter!: ErrorAdapterInterface;
+	private validationAdapter!: ValidationAdapterInterface;
+	private storageAdapter!: StorageAdapterInterface;
+	private globalAuthConfig!: object;
+	private cacheAdapter!: CacheAdapterInterface;
+	private twoFactorProvider!: TwoFactorProviderInterface;
+	// private twoFactorProviders: Map<string, TwoFactorProviderInterface>;
```

Comment:
```suggestion
        // TODO get back to this when we add multiple twoFactorProviders
	// private twoFactorProviders: Map<string, TwoFactorProviderInterface>;
```
Maybe add a TODO so we don't have commented lines dangling like this.